lib/adapter.js

```diff
@@ -5593,11 +5593,16 @@ function Adapter(options) {
                 callback && callback('States database not connected');
                 return;
             }
+
             if (!adapterObjects) { // if objects is no longer existing, we do not need to unsubscribe
                 callback && callback('Objects database not connected');
                 return;
             }
 
+            if (typeof id !== 'string') {
+                return callback(`ID can be only string and not "${typeof id}"`);
```

Comment:
You should check that `callback` exists (like in the lines above). Same goes for all other places where you added this.

Reply:
https://github.com/ioBroker/ioBroker.js-controller/pull/696/files#diff-a7bcb5fcc46018435e1ce3fa65fcb992L5588

Reply:
Hmm. In that case, the other checks could be removed.